src/TNodeImp.h

```diff
@@ -0,0 +1,80 @@
+//
+// Created by shvmo on 29/04/2020.
+//
+
+#ifndef DATASTRUCTURESWET1_TNODEIMP_H
+#define DATASTRUCTURESWET1_TNODEIMP_H
+
+
+#include "TNode.h"
+
+using std::make_shared;
+
+template<class Key, class Info>
+TNode<Key, Info>::TNode() = default;
+
+template<class Key, class Info>
+TNode<Key, Info>::~TNode() = default;
+
+template<class Key, class Info>
+Key &TNode<Key, Info>::getKey() {
+    return this->key;
+}
+
+template<class Key, class Info>
+Info &TNode<Key, Info>::getInfo() {
+    return *(this->infoPointer.get());
+}
+
+template<class Key, class Info>
+shared_ptr<Info> TNode<Key, Info>::getInfoPointer() const {
+    return this->infoPointer;
+}
+
+template<class Key, class Info>
+shared_ptr<TNode<Key, Info>> TNode<Key, Info>::getLeft() const {
+    return this->left;
+}
+
+template<class Key, class Info>
+shared_ptr<TNode<Key, Info>> TNode<Key, Info>::getRight() const {
+    return this->right;
+}
+
+template<class Key, class Info>
+void TNode<Key, Info>::setKey(const Key &newKey) {
+    this->key = newKey;
+}
+
+template<class Key, class Info>
+void TNode<Key, Info>::setInfo(const Info &newInfo) {
+    this->infoPointer = make_shared<Info>(newInfo);
+}
+
+template<class Key, class Info>
+void TNode<Key, Info>::setInfoPointer(shared_ptr<Info> newInfoPointer) {
+    this->infoPointer = newInfoPointer;
+}
+
+
+template<class Key, class Info>
+void TNode<Key, Info>::setLeft(shared_ptr<TNode<Key, Info>> newLeft) {
+    this->left = newLeft;
+}
+
+template<class Key, class Info>
+void TNode<Key, Info>::setRight(shared_ptr<TNode<Key, Info>> newRight) {
+    this->right = newRight;
+}
+
+template<class Key, class Info>
+int TNode<Key, Info>::getHeight() const {
```

Comment:
if you moved the getter in the header move here too plz

Reply:
What? There are no getters in the header?

Reply:
fiiixed